### ISSUE TITLE
Structured Responses

### ISSUE THREAD

**Issue by johnisanerd** [andrewyng/aisuite#66]
I would like to suggest you support structured responses, in particular JSON responses.  Are there any plans for structured responses?

**Comment by GarfieldHuang** [comment]
I also need a similar function to beta.chat.completions.parse. Thank you very much!

`client = OpenAI(api_key=openai_api_key)

def get_jsoncompletion(messages, model="gpt-4o", temperature=0, max_tokens=4000):
    completion = client.beta.chat.completions.parse(
        model="gpt-4o",
        messages=[
            {"role": "system", "content": "You are a helpful GPT. Answer the user the solution step by step."},
            {"role": "user", "content": messages},
        ],
        response_format=LLM_Format_Output,
    )

    jsonCompletion = completion.choices[0].message.parsed
    return jsonCompletion`

**Comment by rohitprasad15** [comment]
Thanks for the request.
Did you mean this  https://platform.openai.com/docs/guides/structured-outputs  ?
"Structured outputs" is under `client.beta`, so not supporting it currently.

If you simply want json responses then you could use it  like this -

<img width="1074" alt="aisuite - json response" src="https://github.com/user-attachments/assets/7378cbd0-4548-4cba-a392-e3b06273ea9d">

**Comment by boxabirds** [comment]
Exactly: the two things you probably want to add to a prompt to increase the likelihood of compliance with a structure is the schema, and examples.

**Comment by luisrock** [comment]
Support for a json response enforcement is very important, in my opinion.

**Comment by gbengochea** [comment]
I totally agree, it seems I am forced to workaround this by using tools, but I just want to enforce json format in the response, almost all LLM providers offer this parameter, enthropic, openai and gemini do for sure. But they do it in different ways, so isn't here where the magic of an integration layer like aisuite would shine ?
